### PR TITLE
[TypeDeclaration] Skip possible void and return by docblock on ReturnTypeFromMockObjectRector

### PR DESCRIPTION
**Pull request by samsonasik**
- Use `ReturnAnalyzer` to verify possible void
- Use `getNativeType()` to ensure it returns native type

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by TomasVotruba** [comment]
:+1:

File: rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php [hunk 1]
```diff
@@ -101,12 +103,14 @@ public function refactorWithScope(Node $node, Scope $scope): ?Node
             return null;
         }
 
-        $soleReturn = $returns[0];
-        if (! $soleReturn->expr instanceof Expr) {
+        if (! $this->returnAnalyzer->hasOnlyReturnWithExpr($node, $returns)) {
             return null;
         }
 
-        $returnType = $this->getType($soleReturn->expr);
```
Comment (staabm) [inline rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php:109]: I wonder whethere there are real use-cases for `getType()` anymore, or whether this method should be deprecated in favor of `getNativeType`...?

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php:109]: `getType()` is stil usefull for:

- get type from php native function/methods, yes, it got mixed on getNativeType()
-  downgrade process
- get object type from method call / static call
- verify transformation that too much for only native check, eg,

```php
/** @var string $var */
trim($var)
 ```

that cast `(string)` is not needed on that case.

Comment (staabm) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php:109]: Ok, said differently: its only useful in very few edge cases but the name misleads even rector-src core developers to use it wrongly in lots of situations

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php:109]: Given rare context, this is used on purpose here 👍

**Pull request merged**